packages/experience-editor/src/components/form/checkbox.tsx

```diff
@@ -12,6 +12,12 @@ export interface CheckboxInputProps {
 export const CheckboxInput: React.FC<CheckboxInputProps> = (textInputProps) => {
   const { field, formValues, handleChange, visible } = textInputProps;
 
+  useEffect(() => {
+    if (!visible) {
+      handleChange(field.id, "false");
+    }
+  }, [visible]);
```

Comment:
do we need `field.id` in the dependency array here? maybe not since it should be stable